Graficar la region de factibilidad y encontrar el mínimo global a partir de las KKT:
\begin{align*}
			\underset{\boldsymbol{x}}{ \text{minimize } } & (x_1-9)^2+(x_2-4)^2,\\
			\text{subject to } & -x_1+3x_2 \leq 12,\\
			& -x_1+x_2 \leq 8,\\
			& 2x_1-x_2 \leq 10,\\
			&x_1,x_2 \geq 0.
		\end{align*}

Organizar ecuaciones
\begin{align*}
			\underset{\boldsymbol{x}}{ \text{minimize } } & (x_1-9)^2+(x_2-4)^2,\\
			\text{subject to } & -x_1+3x_2 - 12 \leq 0 , \ c_1(\boldsymbol{x})\leq 0\\
			& -x_1+x_2 - 8 \leq 0, \ c_2(\boldsymbol{x})\leq 0\\
			& 2x_1-x_2 - 10 \leq 0, c_3(\boldsymbol{x})\leq 0\\
			& -x_1 \leq 0, \ c_4(\boldsymbol{x})\leq 0 \\
      & -x_2 \leq 0, \ c_5(\boldsymbol{x})\leq 0
\end{align*}

$$ L(\mathbf{x},\boldsymbol{\lambda}) = (x_1-9)^2+(x_2-4)^2 + \lambda_1(-x_1+3x_2 - 12) + \lambda_2(-x_1+x_2 - 8) + \lambda_3(2x_1-x_2 - 10) + \lambda_4(-x_1) +  \lambda_5(-x_2)$$

Las condiciones KKT son las siguientes:

1) Estacionariedad
$$ \nabla_\mathbf{x} L(\mathbf{x},\boldsymbol{\lambda}) = \begin{bmatrix} 2(x_1-9) -\lambda_1 -\lambda_2 +2\lambda_3-\lambda_4 \\
2(x_2-4) + 3\lambda_1 + \lambda_2 - \lambda_3 - \lambda_5
\end{bmatrix} = \begin{bmatrix} 0\\ 0
\end{bmatrix}
$$

3) Factibilidad primaria
\begin{align*}
			-x_1+3x_2 - 12 \leq 0 &, \ c_1(\boldsymbol{x})\leq 0\\
			-x_1+x_2 - 8 \leq 0 &, \ c_2(\boldsymbol{x})\leq 0\\
			2x_1-x_2 - 10 \leq 0 &, c_3(\boldsymbol{x})\leq 0\\
			-x_1 \leq 0 &, \ c_4(\boldsymbol{x})\leq 0 \\
      -x_2 \leq 0 &, \ c_5(\boldsymbol{x})\leq 0
\end{align*}

4) Factibilidad dual
$$ \lambda_1 \geq 0, \lambda_2 \geq 0, \lambda_3 \geq 0, \lambda_4 \geq 0, \lambda_5 \geq 0. $$

5. Holgura complementaria (conjunto activo)
\begin{align*}
			\lambda_1(-x_1+3x_2 - 12) = 0 &,\\
			\lambda_2(-x_1+x_2 - 8) = 0 &, \\
			\lambda_3(2x_1-x_2 - 10) = 0 &,\\
			\lambda_4(-x_1) = 0 &, \\
      \lambda_5(-x_2) = 0 &,
\end{align*}

Grafica de la region de factibilidad
$$x_2 = 4+\frac{1}{3}x_1 $$
Asumir $x_1=0$, entonces $x_2=4$.

Asumir $x_2=0$, entonces $x_1=-12$.

# Interseccion 1 y 3

Analizamos la interseccion entre las restricciones 1 y 3, entonces las restricciones activas son la 1 y 3. Los $\lambda$'s de las restricciones 2, 4 y 5 son 0.



$$ \begin{align} 2(x_1-9) -\lambda_1 +2\lambda_3 = 0\\
2(x_2-4) + 3\lambda_1  - \lambda_3 = 0
\end{align}
$$

\begin{align*}
			\lambda_1(-x_1+3x_2 - 12) = 0 &,\\
			\lambda_3(2x_1-x_2 - 10) = 0 &,\\
\end{align*}

$5x_1 =42$; $x_1 =42/5$

$2x_1 - 10=x_2$; $x_2 = 2(42/5)-10=6.8$

In [ ]:
2*(42/5)-10

6.800000000000001

$$ \begin{align} 2(x_1-9)  +2\lambda_3 = \lambda_1\\
2(x_2-4) + 3(2(x_1-9)  +2\lambda_3)  - \lambda_3 = 0
\end{align}
$$

$2(x_2-4) + 6(x_1-9)  +6\lambda_3  - \lambda_3 = 0 $

$5\lambda_3 = -2(x_2-4) - 6(x_1-9)$

$\lambda_3 = -\frac{2}{5}(x_2-4) - \frac{6}{5}(x_1-9)$

In [ ]:
x1 = 42/5
x2 = 6.8
lam3 = -2/5*(x2-4) - 6/5*(x1-9)
lam1 = 2*(x1-9)  +2*lam3

In [ ]:
lam1, lam3

(-2.0, -0.40000000000000036)

El lambda 1 y lambda 3 son negativos, no cumplen con la factiblidad dual. No es un minino del problema.

# Sobre la restriccion 3

Como la restriccion 3 es la unica activa, los lambdas de las restricciones 1, 2, 4 y 5 son 0. El sistema de ecuacione se reduce a:

\begin{align*}
  2(x_1-9)  +2\lambda_3 = 0\\
  4(x_2-4) - 2\lambda_3 = 0\\
	2x_1-x_2 - 10 = 0 \\
\end{align*}

Suma las 2 ecuaciones iniciales para elimiar lambda 3:

\begin{align*}
  2x_1 +4x_2-34  = 0\\
	2x_1-x_2 - 10 = 0 \\
\end{align*}

Restando las 2 ecuaciones, obtenemos:

$$5x_2 = 24, \ x_2=24/5 $$

Despejamos x1 en la segunda ecuacion, en funcion de x2:

$$x_1 = (x_2 + 10)/2$$

In [ ]:
x1 = (24/5+10)/2
x2 = 24/5
x1, x2

(7.4, 4.8)

In [ ]:
lam3_1 = -x1+9
lam3_2 = 2*(x2-4)
lam3_1, lam3_2

(1.5999999999999996, 1.5999999999999996)

El punto x1=7.4 y x2=4.8 es el minimo factible porque el lambda 3 es positivo.